engines/osxnative/sandbox.go

```diff
@@ -0,0 +1,68 @@
+package macengine
+
+import (
+	"github.com/taskcluster/taskcluster-worker/engines"
+	"github.com/taskcluster/taskcluster-worker/runtime"
+	"io"
+	"os"
+	"os/exec"
+)
+
+type sandbox struct {
+	context     *runtime.TaskContext
+	taskPayload *payload
+	env         *[]string
+	aborted     bool
+}
+
+func newSandbox(context *runtime.TaskContext, taskPayload *payload, env *[]string) *sandbox {
+	return &sandbox{
+		context,
+		taskPayload,
+		env,
+		false,
+	}
+}
+
+func (s *sandbox) WaitForResult() (engines.ResultSet, error) {
+	if s.aborted {
+		return resultset{success: false}, engines.ErrSandboxAborted
+	}
+
+	// Use the host environment plus custom environment variables
+	env := os.Environ()
```

Comment:
yea what worried me is if we had a secret like SOME_API_SERVICE_TOKEN that the host is using to communicate somewhere and now that is leaked into the task.  The task environment really needs to be as isolated from anything from the host as possible.